### PR TITLE
fix(#4450 navbarlinks): rotating 'Getting Started' arrow on click

### LINKED ISSUE #4450: 🐞 Bug: Getting started dropdown icon is not working
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Getting started dropdown icon is not working(i.e. it is changing direction of arrow on dropdown)
Screenshot 2023-12-29 060736
Screenshot 2023-12-29 060750

### Steps To Reproduce

1. Go to main page 
2. click on Getting started button 
3. its dropdown icon is not working(i.e. it is changing direction of arrow on dropdown)

### Expected Behavior

IN DROP DOWN IT SHOULD POINT DOWNWARDDS LIKE THIS:
Screenshot 2023-12-29 061353
Screenshot 2023-12-29 061411

### Screenshots

_No response_

### Device Information [optional]

```markdown
- OS:
- Browser:
- version:
```

### Additional context

_No response_

### Are you working on this issue?

Yes

### PR DESCRIPTION
**Pull request by priyanshagra**
Fixes #4450

### Screenshots of the changes (If any) -

https://github.com/CircuitVerse/CircuitVerse/assets/127022024/fc5cebc5-b194-4896-9ee1-342186a00d17

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Review by tanmoysrt — CHANGES_REQUESTED** [review]
Hi @priyanshagra 
I have a doubt.

You have added click event listener to toggle the arrow direction but after opening the menu, if you click anywhere on the page, the menu will be hide again. With this approach that may not change direction of arrow in all cases.

WDYT ??

**Comment by priyanshagra** [comment]
@tanmoysrt changes are fixed please check pr 

https://github.com/CircuitVerse/CircuitVerse/assets/127022024/444b79e1-0c0b-4eab-ae35-ec33c54e3ee0

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -68,6 +68,22 @@
...
+    else
+    {
+        gettingStartedLink.classList.toggle("rotate180");
+    }
+  });
+ document.addEventListener("click", function () {
+    if (gettingStartedLink.classList.contains("rotate180")) {
+      gettingStartedLink.classList.remove("rotate180");
+    } });
+});
```
Comment (tanmoysrt) [inline app/views/layouts/_header.html.erb:86]: 1. There seems to be no element with `navbar-dropdown-2`
2. Give a meaningful id, maybe like `getting-started-dropdown` or anything better
3. properly format the code to match the code style

**Review by tanmoysrt — CHANGES_REQUESTED** [review]

File: app/views/layouts/_header.html.erb (hunk 1 above)
Comment (priyanshagra) [reply app/views/layouts/_header.html.erb]: sir changes done

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -68,6 +68,22 @@
...
+  document.addEventListener("DOMContentLoaded", function () {
+    var gettingStartedLink = document.querySelector("#getting-started-dropdown");
+    gettingStartedLink.addEventListener("click", function () {
+      if (gettingStartedLink.classList.contains("rotate180")) {
+        gettingStartedLink.classList.remove("rotate180");
+      }
+      else{
+        gettingStartedLink.classList.toggle("rotate180");
+      }
+    });
```
Comment (tanmoysrt) [inline app/views/layouts/_header.html.erb:80]: `toggle` method itself add class if not present and remove if present, so it's not required to manually remove if contains
Ref - https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle

**Comment by priyanshagra** [comment]
@tanmoysrt please check this also

**Comment by priyanshagra** [comment]
@tanmoysrt @tachyons please check

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -70,6 +70,17 @@
...
 <script type="text/javascript">
+  document.addEventListener("DOMContentLoaded", function () {
+    var gettingStartedLink = document.querySelector("#getting-started-dropdown");
+    gettingStartedLink.addEventListener("click", function () {
+      gettingStartedLink.classList.toggle("rotate180");
+    });
+    document.addEventListener("click", function () {
+     if (gettingStartedLink.classList.contains("rotate180")) {
+       gettingStartedLink.classList.remove("rotate180");
+     }
```
Comment (tanmoysrt) [inline app/views/layouts/_header.html.erb:81]:
```suggestion
     gettingStartedLink.classList.remove("rotate180");
```
You don't need to check if it contains, as .remove method doesn't raise error if class not exists

**Review by tanmoysrt — APPROVED** [review]
Looks good :+1:

**Pull request merged**